aces/Loris
# [Modules] Sites/Project/subproject filters populated according to user's permission

This PR:
1. modifies the way project/subproject/site filters are populated. Now those filters are displayed with values the current user has permission for.
2. fixed the data table for some modules, to remove entries for Projects and Sites the user does not have access to.
3. add some integration tests

## To test
* Visit all the modules and check that all data entries belong to sites and projects the user has access to.
* Visit all the modules and check that the site/project/subproject menu filters contain entries the user has access to.
* Modified modules:
    - behavioural_qc 
    - candidate_list
    - media
    - new_profile
    - statistics
    - user_accounts
    - survey_accounts

Related to #6829
Related to #6729
